flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/scalar/ArraySortFunction.java

```diff
@@ -0,0 +1,127 @@
+
+/** Implementation of ARRAY_SORT function. */
+@Internal
+public class ArraySortFunction extends BuiltInScalarFunction {
+
+    private final ArrayData.ElementGetter elementGetter;
+    private final SpecializedFunction.ExpressionEvaluator greaterEvaluator;
+
+    private transient MethodHandle greaterHandle;
+
+    public ArraySortFunction(SpecializedFunction.SpecializedContext context) {
+        super(BuiltInFunctionDefinitions.ARRAY_SORT, context);
+        final DataType elementDataType =
+                ((CollectionDataType) context.getCallContext().getArgumentDataTypes().get(0))
+                        .getElementDataType()
+                        .toInternal();
+        elementGetter =
+                ArrayData.createElementGetter(elementDataType.toInternal().getLogicalType());
+        greaterEvaluator =
+                context.createEvaluator(
+                        $("element1").isGreater($("element2")),
+                        DataTypes.BOOLEAN(),
+                        DataTypes.FIELD("element1", elementDataType.notNull().toInternal()),
+                        DataTypes.FIELD("element2", elementDataType.notNull().toInternal()));
+    }
+
+    @Override
+    public void open(FunctionContext context) throws Exception {
+        greaterHandle = greaterEvaluator.open(context);
+    }
+
+    public @Nullable ArrayData eval(ArrayData array) {
+        return eval(array, true, true);
+    }
+
+    public @Nullable ArrayData eval(ArrayData array, Boolean ascendingOrder) {
+        return eval(array, ascendingOrder, ascendingOrder);
+    }
+
+    public @Nullable ArrayData eval(ArrayData array, Boolean ascendingOrder, Boolean nullFirst) {
+        try {
+            if (array == null || ascendingOrder == null || nullFirst == null) {
+                return null;
+            }
+            if (array.size() == 0) {
+                return array;
+            }
+            Object[] elements = new Object[array.size()];
+            for (int i = 0; i < array.size(); i++) {
+                elements[i] = elementGetter.getElementOrNull(array, i);
+            }
+            Comparator<Object> ascendingComparator = new ArraySortComparator(ascendingOrder);
+            Arrays.sort(
+                    elements,
+                    nullFirst
+                            ? Comparator.nullsFirst(ascendingComparator)
+                            : Comparator.nullsLast(ascendingComparator));
+            return new GenericArrayData(elements);
+        } catch (Throwable t) {
+            throw new FlinkRuntimeException(t);
+        }
+    }
+
+    private class ArraySortComparator implements Comparator<Object> {
+        private final boolean isAscending;
+
+        public ArraySortComparator(boolean isAscending) {
+            this.isAscending = isAscending;
+        }
+
+        @Override
+        public int compare(Object o1, Object o2) {
+            if (o1 == null || o2 == null) {
+                return 0;
+            }
```

Comment:
Is this necessary? This should be handled by the `nullsFirst/nullsLast`, no?

Reply:
I check it, it not necessary, the IDE suggest `Comparator does not return 0 for equal elements` so I add it, consider we use isGreater here, so we can remove it.